baseclasses/BaseSolver.py

```diff
@@ -131,16 +72,30 @@ def setOption(self, name, value):
         if self.solverCreated and name in self.imOptions:
             raise Error("Option '%-35s' cannot be modified after the solver " "is created." % name)
 
-        # Now we know the option exists, lets check if the type is ok:
-        if isinstance(value, self.defaultOptions[name][0]):
-            # Just set:
-            self.options[name] = [type(value), value]
+        # If the default provides a list of acceptable values, check whether the value is valid
+        if isinstance(default[1], list) and default[0] is not list:
+            if value in default[1]:
+                self.options[name] = [type(value), value]
```

Comment:
Any idea why we store the type in `self.options`? That information is already stored in `self.defaultOptions` right? Isn't it redundant here?

Reply:
That's a good question. The type in `self.options` is not used in BaseSolver. It's possible that it's used directly in another code though, so changing it might break something.

Reply:
I can do some digging with grep to see if anything else uses it, and then also check with Docker.

Reply:
I did a quick check of our codes that use BaseSolver, and it doesn't look like any of them directly access `.options`.

Reply:
Cool, shall we remove these then?